webhooks/virtualmachinesnapshot/validation/virtualmachinesnapshot_validator.go

```diff
@@ -56,22 +61,85 @@ func (v validator) For() schema.GroupVersionKind {
 	return vmopv1.GroupVersion.WithKind(reflect.TypeOf(vmopv1.VirtualMachineSnapshot{}).Name())
 }
 
+// ValidateCreate makes sure the VirtualMachineSnapshot create request is valid.
 func (v validator) ValidateCreate(ctx *pkgctx.WebhookRequestContext) admission.Response {
-	// TODO: Implement validation for create
+	vmSnapshot, err := v.vmSnapshotFromUnstructured(ctx.Obj)
+	if err != nil {
+		return webhook.Errored(http.StatusBadRequest, err)
+	}
 
 	validationErrs := make([]string, 0)
 
+	if vmSnapshot.Spec.VMRef == nil {
+		validationErrs = append(validationErrs, "spec.vmRef must not be nil")
+	}
+
+	if vmSnapshot.Spec.VMRef != nil {
+		if vmSnapshot.Spec.VMRef.Kind != virtualMachineKind {
+			validationErrs = append(validationErrs, fmt.Sprintf("spec.vmRef.kind must be %q, got %q",
+				virtualMachineKind, vmSnapshot.Spec.VMRef.Kind))
+		}
+
+		if vmSnapshot.Spec.VMRef.APIVersion == "" {
```

Comment:
Is this something that we really have to require? Can't we just implicitly default this (no need for a mutation webhook imo)

Reply:
Default in the controller?

In future, if `v1alpha5` and further comes out and there's some API change that makes the controller behave differently, then this the APIVersion is important to be given by the users.

Reply:
AFACT current we don't ever use the APIVersion value of this field. This validation doesn't really add much value - just not empty but could be a bogus value.

I think for most situation the APIVersion doesn't really matter. Maybe someday we'll use that as an indication of what version of the yaml to store or something but if unset it is reasonable to just assume whatever the current storage version is.